### PR TITLE
use blob names instead of hashes

### PR DESCRIPTION
**Pull request by rogpeppe**
We can't use content addressing with juju blob store,
as discussed ad nauseam online.

This is a redo of https://github.com/juju/charmstore/pull/71,
which had accidentally pulled in some code from another
branch.

### CODE REVIEW

File: internal/charmstore/store_test.go [hunk 1]
```diff
@@ -56,9 +56,12 @@ func (s *StoreSuite) checkAddCharm(c *gc.C, ch charm.Charm) {
 
 	doc.UploadTime = time.Time{}
 
+	blobName := doc.BlobName
+	c.Assert(blobName, gc.Matches, "[0-9a-z]+")
+	doc.BlobName = ""
 	c.Assert(doc, jc.DeepEquals, mongodoc.Entity{
-		URL:                     url,
-		BaseURL:                 mustParseReference("cs:wordpress"),
+		URL:     url,
```
Comment (frankban) [inline internal/charmstore/store_test.go]: Is this gofmted?

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/119/

**Comment by frankban** [comment]
:+1:

File: internal/charmstore/store_test.go (hunk 1 above)
Comment (rogpeppe) [inline internal/charmstore/store_test.go]: no. done.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/120/

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**